db/ActionService/ActionModel.js

```diff
@@ -0,0 +1,25 @@
+const assert = require("assert");
+
+class ActionModel {
```

Comment:
Have you tried using immutable.record for this. Seems like that cut down on some of the boilerplate